packages/core/src/NonceValidator.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Interface for a class which can validate nonces to ensure that the same nonce isn't re-used between requests.
+ */
+export interface NonceValidator {
```

Comment:
I think we can drop this interface declaration unless we plan to extend with multiple classes in the future. One nice thing about TypeScript classes is you can just import its type from a different file like `import { type InMemoryNonceValidator } from "..."` and everything will be typed for you automatically. That way you don't have to maintain the parallel interface. Nbd, just a suggestion!

Reply:
This actually does need to be an interface because it's part of the public-facing SDK API surface and it'll be overridden by implementers to use whatever storage mechanism they want for nonces (probably a persistent db in practice rather than in memory).